pkg/services/search/service.go

```diff
@@ -14,7 +15,7 @@ import (
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-func ProvideService(cfg *setting.Cfg, sqlstore db.DB, starService star.Service, dashboardService dashboards.DashboardService) *SearchService {
+func ProvideService(cfg *setting.Cfg, sqlstore db.DB, starService star.Service, dashboardService dashboards.DashboardService, tracer tracing.Tracer) *SearchService {
```

Comment:
Can we use a package-level tracer instead? IMO it's a much simpler pattern to use, and we get a useful component name out of it:

```suggestion
var tracer = otel.Tracer("github.com/grafana/grafana/pkg/services/search")

func ProvideService(cfg *setting.Cfg, sqlstore db.DB, starService star.Service, dashboardService dashboards.DashboardService) *SearchService {
```